notebooks/customizing_potentials_cookbook.ipynb

```diff
@@ -0,0 +1,1780 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "customizing_potentials_cookbook.ipynb",
+      "provenance": [],
+      "private_outputs": true,
+      "collapsed_sections": [
+        "tYUUT2SvD6B3",
+        "bvp_-L3cJYiM"
+      ],
+      "toc_visible": true
+    },
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    },
+    "accelerator": "GPU"
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "tYUUT2SvD6B3"
+      },
+      "source": [
+        "### Imports & Utils"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "kNDQ02qlDnNW",
+        "colab": {}
+      },
+      "source": [
+        "!pip install --upgrade -q https://storage.googleapis.com/jax-releases/cuda$(echo $CUDA_VERSION | sed -e 's/\\.//' -e 's/\\..*//')/jaxlib-$(pip search jaxlib | grep -oP '[0-9\\.]+' | head -n 1)-cp36-none-linux_x86_64.whl\n",
+        "!pip install --upgrade -q jax\n",
+        "!pip install -q git+https://www.github.com/google/jax-md"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "y9jcXj44Bvcj",
+        "colab": {}
+      },
+      "source": [
+        "import numpy as onp\n",
+        "\n",
+        "import jax.numpy as np\n",
+        "from jax import random\n",
+        "from jax import jit, grad, vmap, value_and_grad\n",
+        "from jax import lax\n",
+        "from jax import ops\n",
+        "\n",
+        "from jax.config import config\n",
+        "config.update(\"jax_enable_x64\", True)\n",
+        "\n",
+        "from jax_md import space, smap, energy, minimize, quantity, simulate, partition\n",
+        "\n",
+        "from functools import partial\n",
+        "\n",
+        "f32 = np.float32\n",
+        "f64 = np.float64\n",
+        "\n",
+        "import matplotlib\n",
+        "import matplotlib.pyplot as plt\n",
+        "plt.rcParams.update({'font.size': 16})\n",
+        "#import seaborn as sns \n",
+        "#sns.set_style(style='white')\n",
+        "\n",
+        "def format_plot(x, y):  \n",
+        "  plt.grid(True)\n",
+        "  plt.xlabel(x, fontsize=20)\n",
+        "  plt.ylabel(y, fontsize=20)\n",
+        "  \n",
+        "def finalize_plot(shape=(1, 0.7)):\n",
+        "  plt.gcf().set_size_inches(\n",
+        "    shape[0] * 1.5 * plt.gcf().get_size_inches()[1], \n",
+        "    shape[1] * 1.5 * plt.gcf().get_size_inches()[1])\n",
+        "  plt.tight_layout()\n",
+        "\n",
+        "def run_minimization(energy_or_force, R_init, shift, num_steps=5000, quant=quantity.Energy):\n",
```

Comment:
It might be better, to define this function (or at least the version without `quant`) in the body of the notebook. You do this for the brownian motion code and I think makes things seem a bit more explicit.

Reply:
Check. For now I have defined a version without ```quant``` in the body and left this version (now called ```run_minimization_general``` in the Utils section. I might move this down to the point where it is needed in the specifying forces section...

Reply:
Update: I moved all the minimization and simulation routines out of the Utils section. I think this is more clear.